File: tests/parsers/samples/code/jl/simple.jl
module SimpleMetrics

using ..ComplexMetrics

export run_simple, summarize

function run_simple()
    settings = ComplexMetrics.build_default()
    baseline = ComplexMetrics.distance(10, 3; scale = settings.scale)
    rms = ComplexMetrics.root_mean_square([1.0, 2.0, 3.0])
    (; baseline, rms)
end

function summarize()
    metrics = run_simple()
    "baseline=$(round(metrics.baseline, digits=2)), rms=$(round(metrics.rms, digits=2))"
end

end

File: tests/parsers/samples/code/lua/simple.lua
local complex = require("complex")

local simple = {}

function simple.run_simple()
    local settings = complex.build_default()
    local baseline = complex.distance(10, 3) * settings.scale
    local rms = complex.root_mean_square({ 1, 2, 3 })
    return { baseline = baseline, rms = rms }
end

function simple.summarize()
    local metrics = simple.run_simple()
    return string.format("baseline=%.2f, rms=%.2f", metrics.baseline, metrics.rms)
end

return simple
